tt_metal/impl/dispatch/command_queue.cpp

```diff
@@ -1144,6 +1244,55 @@ void EnqueueProgramImpl(CommandQueue& cq, std::variant < std::reference_wrapper<
     }, program);
 }
 
+void EnqueueQueueRecordEvent(CommandQueue& cq, Event &event) {
+    TT_ASSERT(cq.id() == 0, "EnqueueQueueRecordEvent only supported on first command queue on device for time being.");
```

Comment:
Is this restriction necessary? Why can't we support recording events on CQ1?

Reply:
Not necessary, but keeping until I can test with multi-CQ (WIP today, hope to include with this PR).